test/simulator/misc/statuses.js

```diff
@@ -0,0 +1,89 @@
+var assert = require('assert');
+var battle;
+
+describe('Burn', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should inflict 1/8 of max HP at the end of the turn, rounded down', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Machamp', ability: 'noguard', moves: ['bulkup']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Sableye', ability: 'prankster', moves: ['willowisp']}]);
+		battle.commitDecisions();
+		var pokemon = battle.p1.active[0];
+		assert.strictEqual(pokemon.maxhp - pokemon.hp, Math.floor(pokemon.maxhp / 8));
+	});
+
+	it('should halve damage from most Physical attacks', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Machamp', ability: 'noguard', moves: ['boneclub']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Sableye', ability: 'prankster', moves: ['splash', 'willowisp']}]);
+		battle.seed = [0, 0, 0, 0];
+		battle.commitDecisions();
+		var pokemon = battle.p2.active[0];
+		var damage = pokemon.maxhp - pokemon.hp;
+		pokemon.hp = pokemon.maxhp;
+		battle.seed = [0, 0, 0, 0];
+		battle.choose('p2', 'move 2');
+		battle.commitDecisions();
+		assert.strictEqual(pokemon.maxhp - pokemon.hp, battle.modify(damage, 0.5));
+	});
+
+	it('should not halve damage from moves with set damage', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Machamp', ability: 'noguard', moves: ['seismictoss']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Talonflame', ability: 'galewings', moves: ['willowisp']}]);
+		battle.commitDecisions();
+		assert.strictEqual(battle.p2.active[0].maxhp - battle.p2.active[0].hp, 100);
+	});
+});
+
+describe('Paralysis', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should reduce speed to 25% of its original value', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Vaporeon', ability: 'waterabsorb', moves: ['aquaring']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Jolteon', ability: 'voltabsorb', moves: ['thunderwave']}]);
+		var speed = battle.p1.active[0].getStat('spe');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].getStat('spe'), battle.modify(speed, 0.25));
+	});
+});
+
+describe('Toxic Poison', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should inflict (active turns with Toxic)/16 of max HP at the end of the turn, rounded down', function () {
```

Comment:
RIP

I switched it to Chansey, which has 622 max HP and isn't divisible by 16.